clippy_lints/src/unit_types/utils.rs

```diff
@@ -0,0 +1,10 @@
+use rustc_hir::{Expr, ExprKind};
+use rustc_middle::ty::{self, Ty};
+
+pub(super) fn is_unit(ty: Ty<'_>) -> bool {
+    matches!(ty.kind(), ty::Tuple(slice) if slice.is_empty())
+}
```

Comment:
No need for this function, since [`TyS::is_unit`](https://doc.rust-lang.org/nightly/nightly-rustc/rustc_middle/ty/struct.TyS.html#method.is_unit) does exactly this check.